packages/turbo/ios/RNVisitableView.swift

```diff
@@ -8,98 +8,88 @@
+  private lazy var turboSession: Session = session.turboSession
+  private lazy var webView: WKWebView = turboSession.webView
+  private lazy var webViewConfiguration: WKWebViewConfiguration = {
+    let configuration = WKWebViewConfiguration()
+    configuration.applicationNameForUserAgent = applicationNameForUserAgent as String?
+    return configuration
+  }()
     
-    guard let sessionModule = self.bridge?.uiManager.moduleRegistry.module(forName: "RNVisitableViewModule") as? RNVisitableViewModule else {
-        print("Couldn't find session for sessionHandle:", sessionHandle ?? "default session")
-        return nil
-    }
-    return sessionModule.getSession(sessionHandle: sessionHandle)
+  lazy var controller: RNVisitableViewController = {
+    let controller = RNVisitableViewController()
+    controller.delegate = self
+    return controller
+  }()
+    
+  override func didMoveToWindow() {
+    reactViewController().addChild(controller)
+    controller.view.frame = bounds // Fixes incorrect size of the webview
+    controller.didMove(toParent: reactViewController())
+    addSubview(controller.view)
   }
-  
-  public func getSession() -> Session? {
-    return getRNSesssion()?.turboSession
+    
+  func attachDelegateAndVisit() {
+    turboSession.delegate = self
+    performVisit()
   }
-  
-  override func didMoveToWindow() {
-
-    let url = URL(string: String(url))!
     
-    if (self.controller == nil) {
-      print("Open new VistableView with URL passed from JS: ", url)
-      setupViewController(url: url)
-      self.addSubview((controller?.view)!)
+  public func handleMessage(message: WKScriptMessage){
+    if let messageBody = message.body as? [AnyHashable : Any] {
+      onMessage?(messageBody)
     }
   }
   
-  func setupViewController(url: URL) {
-    self.controller = RNVisitableViewController(url: url)
-    self.controller?.delegate = self
-    self.reactViewController().addChild(self.controller!)
-    self.controller?.view.frame = bounds // Fixes incorrect size of the webview
-    self.controller?.didMove(toParent: self.reactViewController())
+  public func injectJavaScript(code: NSString) -> Void {
+    webView.evaluateJavaScript(code as String)
   }
-  
-  func attachDelegateAndVisit(_ visitable: Visitable) {
-    let session = getSession()
-    session?.delegate = self
+    
+  private func performVisit(){
     // Upon initial load, the session.visit function is called
     // to set the currentVisit private variable.
-    if(session?.webView.url == nil){
-      session?.visit(visitable)
+    if(webView.url == nil){
+      turboSession.visit(controller)
       return
     }
-    session?.visitableViewWillAppear(visitable)
+    turboSession.visitableViewWillAppear(controller)
+  }
+
+  private func performVisitOnUrlChange(oldUrl: NSString){
+    if(oldUrl != "" && url != "" && oldUrl != url){
+      turboSession.visit(controller, action: .replace)
+    }
```

Comment:
Moved _initial load_ condition to `performVisit` method. I've tried to somehow avoid it but as said in the comment, session.visit function is called to set the currentVisit private variable.

Reply:
I think this is worse than before, haha. Now we have extracted this into a single-use method, that needs a dependent argument (`oldUrl`) to perform it's this.

Reply:
Reason [here](https://github.com/software-mansion-labs/react-native-turbo-demo/pull/57#discussion_r1407767067)